app/Scripts/Player.gd

```diff
@@ -18,7 +18,7 @@ const DashPower = 5
 
 var velocity = Vector2.ZERO
 var Dir: Vector2
-
+var IsIdle = true
```

Comment:
isidle and canmove multiply state and make bugs more likely. ideally we use a state machine here too, but it's fine if this works

Reply:
yeah seemed to work for now, If I find bugs then I'll have to refactor

Reply:
the rats and the boss already have this implemented

Reply:
nvm the rat doesnt lol